src/mlpack/core/optimizers/cmaes/cmaes_impl.hpp

```diff
@@ -0,0 +1,260 @@
+/**
+ * @file cmaes_impl.hpp
+ * @author Marcus Edel
+ * @author Kartik Nighania
+ *
+ * Implementation of the Covariance Matrix Adaptation Evolution Strategy as
+ * proposed by N. Hansen et al. in "Completely Derandomized Self-Adaptation in
+ * Evolution Strategies".
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_IMPL_HPP
+#define MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "cmaes.hpp"
+
+namespace mlpack {
+namespace optimization {
+
+CMAES::CMAES(const size_t lambda,
+             const double lowerBound,
+             const double upperBound,
+             const size_t maxIterations,
+             const double tolerance) :
+    lambda(lambda),
+    lowerBound(lowerBound),
+    upperBound(upperBound),
+    maxIterations(maxIterations),
+    tolerance(tolerance)
+{ /* Nothing to do. */ }
+
+//! Optimize the function (minimize).
+template<typename DecomposableFunctionType>
+double CMAES::Optimize(DecomposableFunctionType& function, arma::mat& iterate)
+{
+  // Find the number of functions to use.
+  const size_t numFunctions = function.NumFunctions();
+
+  // Population size.
+  if (lambda == 0)
+    lambda = (4 + std::round(3 * std::log(iterate.n_elem))) * 10;
+
+  // Parent weights.
+  const size_t mu = std::round(lambda / 2);
+  arma::vec w = std::log(mu + 0.5) - arma::log(arma::conv_to<arma::vec>::from(
+      arma::linspace<arma::uvec>(1, mu, mu + 1)));
```

Comment:
I think you can get away with `arma::linspace<arma::vec>(1, mu, mu + 1)` here, no need for the `conv_to`.

Reply:
Right, I used an incorrect starting point, so that I needed to convert doubles to uword. With the correct starting point I can directly use ``arma::linspace<arma::vec>``.